hangman/lib/hangman/server.ex

```diff
@@ -0,0 +1,54 @@
+defmodule Hangman.Server do
+  use GenServer
+
+  def start_link(state_server) do
+    GenServer.start_link(__MODULE__, 
+                         state_server, 
+                         name: __MODULE__)
+  end
+
+  def init(state_server) do
+    current_game = Hangman.StateServer.get_game(state_server)
+    {:ok, {state_server, current_game}}
+  end
+
+  def new_game() do
+    new_game(Dictionary.random_word)
+  end
+
+  def new_game(word) do
+    GenServer.cast(__MODULE__, {:new_game, word})
+    GenServer.call(__MODULE__, :get_game)
+  end
+
+  def tally(_game) do
+    GenServer.call(__MODULE__, :tally)
+  end
+
+  def make_move(game, guess) do
+    GenServer.cast(__MODULE__, {:make_move, guess})
```

Comment:
# -4

Here's you're mixing levels of code. The cast is a call to the code running in the server, which is responsible for the logic of the game. But then on the next line to run some game logic (`tally`) inline, which kind of defeats the point.

An important thing here is to separate concerns.